components/service/mygeneinfo/index.tsx

```diff
@@ -64,11 +82,32 @@ export const GeneInfoFromGeneTerm = MetaNode('GeneInfoFromGeneTerm')
   .inputs({ gene: GeneTerm })
   .output(GeneInfo)
   .resolve(async (props) => {
-    const results = MyGeneInfoHitC.parse(await mygeneinfo_query(props.inputs.gene))
+    return await getGeneData(props.inputs.gene);
+  })
+  .build()
+
+  export async function getGeneData(geneSymbol: string){
+    const results = MyGeneInfoHitC.parse(await mygeneinfo_query(geneSymbol))
     const hits = results.hits.filter((hit): hit is MyGeneInfoHit['hits'][0] & { symbol: string } => !!hit.symbol)
-    const exactMatch = hits.filter(hit => hit.symbol.toUpperCase() == props.inputs.gene.toUpperCase())[0]
+    const exactMatch = hits.filter(hit => hit.symbol.toUpperCase() == geneSymbol.toUpperCase())[0]
     const _id: string | undefined = exactMatch !== undefined ? exactMatch._id : hits[0]._id
-    if (_id === undefined) throw new Error(`Could not identify a gene for the symbol ${props.inputs.gene} in mygene.info`)
+    if (_id === undefined) throw new Error(`Could not identify a gene for the symbol ${geneSymbol} in mygene.info`)
     return await mygeneinfo(_id)
+  }
+
+
+  export const GetRegulatoryElementsForGeneInfo = MetaNode('GetRegulatoryElementsForGeneInfo')
```

Comment:
I think this should be in a different file/directory, for instance in some "genboree" service directory -- or perhaps it just goes in the regulatoryElementInfo directory